### DISCUSSION TITLE
How to output bash array in prompt

### DISCUSSION THREAD

**Discussion by hwine** [starship/starship#3127] [Q&A] [answered]
pre starship, I started my bash prompt with `${PIPESTATUS[@]}`. Since I often use pipes on the command line, I found this to be helpful in knowing which step(s) threw an error.

So far, I haven't been able to figure out how to replicate this with my starship config.

Any suggestions?

**Comment by andytom** [comment, marked as answer]
You can use the [status module](https://starship.rs/config/#status) to display the pipe status. This isn't enabled by default so you will need to enable it in your configuration.

**Reply by hwine** [reply]
see rest of thread for additional details.

**Comment by hwine** [comment]
@andytom thanks for the pointer, but I can't figure out how to output the `PIPESTATUS` _array_ -- I only get a single value. Let me give a more explicit example of what I'm trying to achieve. With my current manually crafted `PS1` for bash, I get:
```bash
 0 [hwine@hwine-loaner] (master *%=) ~/repos/fx
$ echo fred | false | cat -
 141 1 0 [hwine@hwine-loaner] (master *%=) ~/repos/fx
$
```
There are 3 commands in the pipeline, so there are 3 exit codes displayed. Using `starship`, I appear to only be able to get the value of `$PIPESTATUS` (a single int, the exit code from the last command in the pipeline). How do I get all the array values to be displayed? (In bash, that's `${PIPESTATUS[@]}`.)

I feel like I'm missing something obvious, but a search for "array" in the docs didn't turn up anything relevant. ☹

**Reply by chipbuster** [reply]
Can you confirm that you're trying to get something like this?

![image](https://user-images.githubusercontent.com/4605384/136301259-66b06437-f421-457c-9e49-93d8d29d543a.png)

**Reply by hwine** [reply]
@chipbuster YES!  Can you share the `starship.toml` configuration for that output?  That's what I expected to see, so it's probably some PEBKAC on my end.

**Reply by chipbuster** [reply]
Sure thing. As @andytom mentioned, this is [part of the `status` module](https://starship.rs/config/#status). To enable this, you'll want to have at least

```
[status]
disabled=false
pipestatus=true
```

You can use the other options/variables on the status module page to control how the module behaves (including `pipestatus_format`). Note that the pipestatus code is still fairly new and we're still ironing out bugs + customization issues with it (see #3050), so let us know if there's some behavior you can't seem to get.

**Reply by hwine** [reply]
🎉 That does it -- after I commented out all the other keys under `[status]`. I'd started with the output of `starship print-config`, so something in one of those other settings was interfering.

When I have some time, I'll figure out which setting(s) were causing the issue.
